crates/sp-domains-fraud-proof/src/verification.rs

```diff
@@ -270,3 +272,73 @@ where
 
     Ok(())
 }
+
+pub fn verify_true_invalid_bundle_fraud_proof<CBlock, DomainNumber, DomainHash, Balance>(
+    bad_receipt: ExecutionReceipt<
+        NumberFor<CBlock>,
+        CBlock::Hash,
+        DomainNumber,
+        DomainHash,
+        Balance,
+    >,
+    true_invalid_fraud_proof: &TrueInvalidBundleEntryFraudProof,
+) -> Result<(), VerificationError>
+where
+    CBlock: BlockT,
+{
+    let true_invalid_bundle_entry = bad_receipt
+        .inboxed_bundles
+        .get(true_invalid_fraud_proof.bundle_index as usize)
+        .ok_or(VerificationError::BundleNotFound)?;
+
+    if !true_invalid_fraud_proof.matches_with_bundle_entry(true_invalid_bundle_entry) {
+        return Err(VerificationError::FraudProofMismatch);
+    }
+
+    let extrinsic = StorageProofVerifier::<BlakeTwo256>::get_decoded_value(
+        &true_invalid_bundle_entry.extrinsics_root,
+        StorageProof::new(
+            true_invalid_fraud_proof
+                .extrinsic_inclusion_proof
+                .clone()
+                .drain(..),
+        ),
+        StorageKey(
+            true_invalid_fraud_proof
+                .mismatched_extrinsic_index
+                .to_be_bytes()
+                .to_vec(),
+        ),
+    )
+    .map_err(|_e| VerificationError::InvalidProof)?;
+
+    match true_invalid_fraud_proof.proof_data {
+        ProofDataPerExpectedInvalidBundle::OutOfRangeTx => {
+            let fraud_proof_verification_response =
+                fraud_proof_runtime_interface::get_fraud_proof_verification_info(
+                    H256::from_slice(bad_receipt.consensus_block_hash.as_ref()),
+                    FraudProofVerificationInfoRequest::TxRangeCheck {
+                        domain_id: true_invalid_fraud_proof.domain_id,
+                        consensus_block_hash_with_tx_range: H256::from_slice(
+                            bad_receipt.consensus_block_hash.as_ref(),
+                        ), // TODO: Use correct consensus block hash when dynamic tx range is introduced
+                        consensus_block_tx_range_storage_proof: vec![], // Unused since currently api is stateless
```

Comment:
If this is unused, then remove them rather than passing default value. We can add this back once have a use for it